filebeat/input/filestream/internal/input-logfile/store.go

```diff
@@ -508,7 +508,8 @@ func (r *resource) copyInto(dst *resource) {
 	dst.pendingCursorValue = nil
 	dst.pendingUpdate = nil
 	dst.cursorMeta = r.cursorMeta
-	dst.lock = unison.MakeMutex()
+	// dst.lock should not be overwritten here because it's supposed to be locked
+	// before this function call and it's important to preserve the previous value.
```

Comment:
Is there a regression test you can write for this?

A simple unit test where the mutex is locked before this function call, and then you verify that `dst.lock` is still locked afterwards with `TryLock` would be enough to enforce this behaviour going forward.

Reply:
Added, this is how it would fail if the change was reverted <img width="1044" alt="Screenshot 2022-03-29 at 16 10 57" src="https://user-images.githubusercontent.com/887952/160630908-7a137138-1cbd-4473-8798-81bd7159a0c9.png">